src/test/store/profile-view.test.js

```diff
@@ -116,83 +123,613 @@ describe('call node paths on implementation filter change', function() {
+              hitType: 'event',
+            },
+          ],
+        ]);
+      });
+      expect(UrlStateSelectors.getInvertCallstack(getState())).toEqual(true);
+    });
+  });
+
+  describe('updateProfileSelection', function() {
+    const { profile } = getProfileFromTextSamples('A');
+    const { dispatch, getState } = storeWithProfile(profile);
+
+    it('updates the profile selection', function() {
+      expect(
+        ProfileViewSelectors.getProfileViewOptions(getState()).selection
+      ).toEqual({ hasSelection: false, isModifying: false });
+      dispatch(
+        ProfileView.updateProfileSelection({
+          hasSelection: true,
+          isModifying: false,
+          selectionStart: 0,
+          selectionEnd: 1,
+        })
+      );
+      expect(
+        ProfileViewSelectors.getProfileViewOptions(getState()).selection
+      ).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionStart: 0,
+        selectionEnd: 1,
+      });
+    });
+  });
+
+  describe('addRangeFilter', function() {
+    const { profile } = getProfileFromTextSamples('A');
+    const { dispatch, getState } = storeWithProfile(profile);
+
+    it('adds a range filter', function() {
+      expect(UrlStateSelectors.getRangeFilters(getState())).toEqual([]);
+      dispatch(ProfileView.addRangeFilter(0, 10));
+      expect(UrlStateSelectors.getRangeFilters(getState())).toEqual([
+        { start: 0, end: 10 },
+      ]);
+
+      dispatch(ProfileView.addRangeFilter(1, 9));
+      expect(UrlStateSelectors.getRangeFilters(getState())).toEqual([
+        { start: 0, end: 10 },
+        { start: 1, end: 9 },
+      ]);
+    });
+  });
+
+  describe('addRangeFilterAndUnsetSelection', function() {
+    const { profile } = getProfileFromTextSamples('A');
+    const { dispatch, getState } = storeWithProfile(profile);
+
+    it('adds a range filter and unsets a selection', function() {
+      dispatch(ProfileView.addRangeFilter(0, 10));
+      dispatch(
+        ProfileView.updateProfileSelection({
+          hasSelection: true,
+          isModifying: false,
+          selectionStart: 1,
+          selectionEnd: 9,
+        })
+      );
+      expect(UrlStateSelectors.getRangeFilters(getState())).toEqual([
+        { start: 0, end: 10 },
+      ]);
+      expect(
+        ProfileViewSelectors.getProfileViewOptions(getState()).selection
+      ).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 9,
+        selectionStart: 1,
+      });
+
+      dispatch(ProfileView.addRangeFilterAndUnsetSelection(2, 8));
+      expect(UrlStateSelectors.getRangeFilters(getState())).toEqual([
+        { start: 0, end: 10 },
+        { start: 2, end: 8 },
+      ]);
+      expect(
+        ProfileViewSelectors.getProfileViewOptions(getState()).selection
+      ).toEqual({
+        hasSelection: false,
+        isModifying: false,
+      });
+    });
+  });
+
+  describe('popRangeFilters', function() {
```

Comment:
**optional**: I'd merge this `describe` with the next one. Possibly with a `beforeEach` to factorize our the calls to `addRangeFilter`.

Reply:
Thanks, factorized it with a `setupStore`.